src/hooks/useArNSRecord.tsx

```diff
@@ -0,0 +1,16 @@
+import { useGlobalState } from '@src/state';
+import { useQuery } from '@tanstack/react-query';
+
+export function useArNSRecord({ name }: { name: string }) {
+  const [{ arioContract }] = useGlobalState();
+
+  return useQuery({
+    queryKey: ['arns-record', name],
+    queryFn: () => {
+      if (!name) return undefined;
```

Comment:
useQuery typically returns null for data (serializable), and if throwing an error it catches it and can be accessible as 

```
const {error} = useQuery()
```

```suggestion
      if (!name || !name.length) throw new Error("No ArNS Name provided");
```

Reply:
[c5fc1a7](https://github.com/ar-io/arns-react/pull/854/commits/c5fc1a729c63670f4cb744f046b78cbddbb722bc)